Foundation/NSJSONSerialization.swift

```diff
@@ -68,3 +86,399 @@ public class NSJSONSerialization : NSObject {
+        case UnicodeScalar(0x74): // t    tab             U+0009
+            return (UnicodeScalar(0x09), parser)
+        case UnicodeScalar(0x75): // u    unicode
+            return try parseUnicodeSequence(parser)
+        default:
+            return nil
+        }
+    }
+    
+    static func parseUnicodeSequence(input: UnicodeParser) throws -> (UnicodeScalar, UnicodeParser)? {
+        
+        guard let (codeUnit, parser) = parseCodeUnit(input) else {
+            return nil
+        }
+        
+        if !UTF16.isLeadSurrogate(codeUnit) {
+            return (UnicodeScalar(codeUnit), parser)
+        }
+        
+        guard let (trailCodeUnit, finalParser) = try consumeSequence("\\u", input: parser).flatMap(parseCodeUnit) where UTF16.isTrailSurrogate(trailCodeUnit) else {
+            throw NSJSONSerializationError.MissingTrailingSurrogate(parser.distanceFromStart)
+        }
+        
+        var utf = UTF16()
+        var generator = [codeUnit, trailCodeUnit].generate()
+        switch utf.decode(&generator) {
+        case .Result(let scalar):
+            return (scalar, finalParser)
+        default:
+            return nil
+        }
+    }
+    
+    static let hexScalars = "1234567890abcdefABCDEF".unicodeScalars
+    static func parseCodeUnit(input: UnicodeParser) -> (UTF16.CodeUnit, UnicodeParser)? {
+        guard let (result, parser) = takeInClass(hexScalars, count: 4, input: input),
+            let value = Int(String(result), radix: 16) else {
+                return nil
+        }
+        return (UTF16.CodeUnit(value), parser)
+    }
+    
+    //MARK: - Number parsing
+    static let numberScalars = ".+-0123456789eE".unicodeScalars
+    static func parseNumber(input: UnicodeParser) throws -> (AnyObject, UnicodeParser)? {
+        let view = input.view
+        let endIndex = view.endIndex
+        var index = input.index
+        var value = String.UnicodeScalarView()
+        while index < endIndex && numberScalars.contains(view[index]) {
+            value.append(view[index])
+            index = index.successor()
+        }
+        guard value.count > 0, let result = Double(String(value)) else {
+            return nil
+        }
+        return (result, UnicodeParser(view: view, index: index))
+    }
+
+    //MARK: - Value parsing
+    static func parseValue(input: UnicodeParser) throws -> (AnyObject, UnicodeParser)? {
+        if let (value, parser) = try parseString(input) {
+            return (value, parser)
+        }
+        else if let parser = try consumeSequence("true", input: input) {
+            return (true, parser)
+        }
+        else if let parser = try consumeSequence("false", input: input) {
+            return (false, parser)
+        }
+        else if let parser = try consumeSequence("null", input: input) {
+            return (NSNull(), parser)
+        }
+        else if let (object, parser) = try parseObject(input) {
+            return (object, parser)
```

Comment:
For some reason this is coming out as a return type of ([String : AnyObject], JSONDeserializer.UnicodeParser)
which Dictionary cannot be converted implicitly either.